File: 225-implement-stack-using-queues/implement-stack-using-queues.java
class MyStack {
    ArrayDeque<Integer> first;
    ArrayDeque<Integer> second;
    public MyStack() {
        first = new ArrayDeque();
        second = new ArrayDeque();
    }
    public void push(int x) {
        this.first.add(x);
    }
    
    public int pop() {
        int n = this.first.size();
        for (int i = 0; i < n - 1;i++){
            this.second.add(this.first.poll());
        }
        int popped = this.first.poll();
        while (!this.second.isEmpty()){
            this.first.add(this.second.poll());
        }
        return popped;        
    }
    
    public int top() {
        int n = this.first.size();
        for (int j = 0; j < n - 1; j++){
            this.second.add(this.first.poll());
        }

        int peeked = this.first.poll();
        while (!this.second.isEmpty()){
            this.first.add(this.second.poll());
        }
        this.first.add(peeked);
        return peeked;
    }
    
    public boolean empty() {
        return this.first.isEmpty();
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack obj = new MyStack();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.top();
 * boolean param_4 = obj.empty();
 */

File: 225-implement-stack-using-queues/implement-stack-using-queues.py
class MyStack:

    def __init__(self):
        self.first = deque()
        self.second = deque()       

    def push(self, x: int) -> None:
        self.first.append(x)        

    def pop(self) -> int:
        for i in range(len(self.first) - 1):
            self.second.append(self.first.popleft())
        
        popped = self.first.popleft()

        while len(self.second) != 0:
            self.first.append(self.second.popleft())
        
        return popped       

    def top(self) -> int:
        for i in range(len(self.first) - 1):
            self.second.append(self.first.popleft())
        peeked = self.first.popleft()

        while len(self.second) != 0:
            self.first.append(self.second.popleft())
        self.first.append(peeked)
        return peeked 
        

    def empty(self) -> bool:
        return len(self.first) == 0
        


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()